### PR TITLE
bump syn

### PR DESCRIPTION
**Pull request by burrbull**
Fix some problems in parsing input file.

### CODE REVIEW

**Comment by djmcgill** [comment]
Looks good thanks! Can I ask why you removed the lock file? I was under the impression was that best practice for binary crates was to keep it, to guarantee repeatable behaviour.

**Comment by burrbull** [comment]
> I was under the impression was that best practice for binary crates was to keep it, to guarantee repeatable behaviour.

Sorry, I didn't know you use it.
Should I restore lock file?

**Comment by djmcgill** [comment]
Yes please

**Comment by burrbull** [comment]
Done.

**Comment by djmcgill** [comment]
Your change doesn't seem to be working. After I fixed the compilation error (`syn::parse_item` to `syn::parse_file`) in the test, when running it I got this lib as output:
```
#![cfg_attr(feature = "rt", feature(asm))]
extern crate cortex_m_rt;
use bare_metal::Peripheral;
use core::ops::Deref;
pub const NVIC_PRIO_BITS: u8 = 2;
#[doc(hidden)]
pub mod interrupt {
    pub const INTERRUPT_CONST: u8 = 3;
}
pub use cortex_m::peripheral::TPIU;
pub struct AC;
pub mod ac {
    pub mod ac2 {
        pub const AC2: AC = AC;
        pub mod ac3 {
            pub const AC3: AC = AC;
        }
    }
    pub const AC: AC = AC;
}
```
But that last part is supposed to be just `pub mod ac;`.

I would like to apologise for not having the tests in a CI setup though so you could have seen this immediately

File: src/util.rs [hunk 1]
```diff
@@ -80,61 +81,53 @@ impl<P: AsRef<Path> + Send + Sync> FileIntoMods<P> {
...
-    write_all_tokens(&rust_crate, &mut file)
+    write_all_tokens(&item_mod, &mut file)
 }
 
-fn extract_crate_from_mod<'a>(node: &'a mut ItemKind) -> Option<Crate> {
-    if let ItemKind::Mod(ref mut maybe_items) = *node {
-        maybe_items.take().map(make_crate)
+fn extract_mod<'a>(node: &'a mut Item) -> Option<ItemMod> {
+    if let Item::Mod(mod_item) = &*node {
+        Some(mod_item.clone())
```
Comment (djmcgill) [inline src/util.rs]: I'm pretty sure that this is the problem line - when you fold over a `mod foo { ... }` you need to convert it into `mod foo;`  which the `maybe_items.take()` was doing.

Comment (burrbull) [reply src/util.rs]: Yes, you are right. Sorry again.

**Comment by burrbull** [comment]
I seem to fix it.

**Comment by djmcgill** [comment]
The code looks good, but the tests are still failing with
```
> cargo test
   Compiling form v0.7.0 (/mnt/c/Users/David/Documents/code/form)
error[E0432]: unresolved import `syn::parse_crate`
 --> tests/file-test.rs:7:5
  |
7 | use syn::parse_crate;
  |     ^^^^^-----------
  |     |    |
  |     |    help: a similar name exists in the module: `parse_quote`
  |     no `parse_crate` in the root
```

**Comment by burrbull** [comment]
Oh, I did not see crate have tests. One minute.

**Comment by djmcgill** [comment]
Great!

**Comment by djmcgill** [comment]
Released: https://crates.io/crates/form/0.7.0

**Pull request merged**